addons/addon-base-raas/packages/base-raas-services/lib/study/helpers/entities/study-methods.js

```diff
@@ -0,0 +1,195 @@
+ * and get implicit 'read = true and write = true' for My Studies (if accessType is readwrite).
+ *
+ * @param studyEntity The study entity with the permissions property populated
+ * @param uid The use id
+ */
+function accessLevels(studyEntity, uid) {
+  if (isOpenData(studyEntity)) return { admin: false, read: true, write: false };
+  const permissions = { ...getEmptyStudyPermissions(), ..._.cloneDeep(studyEntity.permissions) };
+  const isAdmin = _.includes(permissions.adminUsers, uid);
+
+  if (isAdmin) {
+    // A study admin will implicitly have read access to the study unless the study accessType
+    // is writeonly, in that case, the study admin will implicitly have write access (unless
+    // it is open data). If the study is 'My Studies' then the admin will also implicitly have
+    // write access (unless it is open data or readonly accessType).
+    permissions.readonlyUsers.push(uid);
+    if (isWriteonly(studyEntity)) permissions.writeonlyUsers.push(uid);
+    if (isMyStudies(studyEntity)) permissions.readwriteUsers.push(uid);
+  }
+
+  // To make the logic easier to implement, we first clear all permissions levels
+  // that are above the maximum level (accessType).
+  // Notice that when we have accessType = readwrite, we don't clear the readonly users
+  // nor the writeonly users, this is because (as mentioned above) accessType presents
+  // the maximum permissions allowed.
+  if (isReadonly(studyEntity)) {
+    // all users who had readwrite access need to be demoted to readonlyUsers
+    if (!_.isEmpty(permissions.readwriteUsers)) {
+      permissions.readonlyUsers = _.uniq([...permissions.readonlyUsers, ...permissions.readwriteUsers]);
+      permissions.readwriteUsers = [];
+    }
+    permissions.writeonlyUsers = [];
+  } else if (isWriteonly(studyEntity)) {
+    // all users who had readwrite access need to be demoted to writeonlyUsers
+    if (!_.isEmpty(permissions.readwriteUsers)) {
+      permissions.writeonlyUsers = _.uniq([...permissions.writeonlyUsers, ...permissions.readwriteUsers]);
+      permissions.readwriteUsers = [];
+    }
+    permissions.readonlyUsers = [];
+  }
+
+  return {
+    admin: isAdmin,
+    read: _.includes(permissions.readonlyUsers, uid) || _.includes(permissions.readwriteUsers, uid),
+    write: _.includes(permissions.writeonlyUsers, uid) || _.includes(permissions.readwriteUsers, uid),
+  };
+}
+
+function isOpenData(studyEntity = {}) {
+  return studyEntity.category === 'Open Data';
+}
+
+function isMyStudies(studyEntity = {}) {
+  return studyEntity.category === 'My Studies';
+}
+
+function isReadonly(studyEntity = {}) {
+  // Due to backward compatibility, not all studyEntity have the property 'accessType', therefore,
+  // by default the accessType if not found is readwrite. Remember the accessType represents the
+  // the max permissions allowed on the study.
+  const accessType = _.get(studyEntity, 'accessType', 'readwrite');
+  return accessType === 'readonly';
+}
```

Comment:
Where do we decide the accessType of the study? Will that happen from the UI going forward?

Reply:
It will only happen when a study is being registered via the api  POST /api/data-sources/accounts/:id/buckets/:name/studies is called.

Existing study creation api is not expected to change.